zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -111,39 +116,95 @@ public InterpreterFactory(ZeppelinConfiguration conf, InterpreterOption defaultO
 
   private void init() throws InterpreterException, IOException, RepositoryException {
```

Comment:
This method now is quite big and may be hard to follow. How do you think, would that be possible/make sense to extract few high-level methods here and make it call them?

It might aid the readability and simplify understanding of new interpreter registration process.

What do you think?